### DISCUSSION TITLE
Allow inline safelist entries

### DISCUSSION THREAD

**Discussion by ntkoopman** [tailwindlabs/tailwindcss#6764] [Ideas]
Right now it's only possible to add safelist entries in the global configuration:

```js
{
  safelist: [ 'sm:hidden', 'md:hidden' ]
}
```

but that makes it difficult to know *why* those classes are included and also easy to forget to remove them if they become unneccesary. It would be great if it was possible to specify safelist entries at the point where they are needed: e.g.

```js
function MyComponent({small) {
  let breakpoint = small ? 'sm': 'md';
  // tailwind-safelist sm:hidden md:hidden -- optional comment
  return <div class={`${breakpoint}:hidden`}>Hello</div>;
}
```

**Comment by adamwathan** [comment]
Hey! That exact code you wrote there will already work, Tailwind will see the classes in the comment and generate them.

Tailwind is deliberately very stupid about detecting class names, it just uses a regex not any sort of parsing so as long as the strings exist they will be generated:

https://tailwindcss.com/docs/content-configuration#class-detection-in-depth

**Comment by ntkoopman** [comment]
Oh no, busted for not trying it out before posting!
